#!/usr/bin/env python
PACKAGE = "vdb_mapping_ros"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("max_range", double_t, 0, "max_range parameter", 10.0, 0.0, 100.0)
gen.add("prob_hit", double_t, 0, "prob_hit parameter", 0.8, 0.0, 1.0)
gen.add("prob_miss", double_t, 0, "prob_miss parameter", 0.1, 0.0, 1.0)
gen.add("prob_thres_min", double_t, 0, "prob_thres_min parameter", 0.12, 0.0, 1.0)
gen.add("prob_thres_max", double_t, 0, "prob_thres_max parameter", 0.8, 0.0, 1.0)

gen.add("publish_pointcloud", bool_t, 0, "publish_pointcloud parameter",  True)
gen.add("publish_vis_marker", bool_t, 0, "publish_vis_marker parameter",  True)
gen.add("publish_updates", bool_t, 0, "publish_updates parameter",  False)

exit(gen.generate(PACKAGE, "vdb_mapping_ros", "VDBMappingROS"))